src/include/index/ivf_pq_index.h

```diff
@@ -1246,15 +1303,17 @@ class ivf_pq_index {
     auto&& [active_partitions, active_queries] =
         detail::ivf::partition_ivf_flat_index<indices_type>(
             flat_ivf_centroids_, query_vectors, nprobe, num_threads_);
+    auto query_centroid_distances = std::move(
+        *pq_query_centroid_distances<Q, ColMajorMatrix<float>>(query_vectors));
     return detail::ivf::query_infinite_ram(
         *partitioned_pq_vectors_,
         active_partitions,
-        query_vectors,
+        query_centroid_distances,
         active_queries,
         k_nn,
         num_threads_,
-        make_pq_distance_asymmetric<
-            std::span<typename Q::value_type>,
+        make_pq_distance_query_centroid_distances<
+            std::span<float>,
```

Comment:
Curious, does `typename Q::value_type` not work anymore?

Reply:
This is always float as it is the distance between the query and the centroid. It is not related to the query vector type anymore.